source/manual/aws-iam-key-rotation.html.md

```diff
@@ -5,93 +5,131 @@ layout: manual_layout
-[Access Keys](https://docs.aws.amazon.com/en_pv/IAM/latest/UserGuide/id_credentials_access-keys.html)
-associated with these accounts need to be [rotated](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_RotateAccessKey)
-every 3 months.
+GOV.UK uses [AWS IAM] accounts to manage applications access to our AWS
+infrastructure. The [Access Keys] associated with these accounts need to be
+[rotated] every 3 months.
 
-## How to rotate Access Keys
+Rotating AWS IAM keys requires permissions which are available with the `admin`,
+`internal-admin`, `platformhealth-poweruser` or `poweruser` roles. If you can't
+assume one of these roles, you won't be able to rotate keys.
 
-> **Note**
->
-> Rotating AWS IAM keys requires permissions which are available with the
-> `admin`, `internal-admin`, `platformhealth-poweruser` or `poweruser` roles.
-> If you can't assume one of these roles, you won't be able to rotate keys.
+## How to rotate Access Keys
 
-Log in to the [IAM Management Console](https://console.aws.amazon.com/iam/home?region=eu-west-1#/users)
-for the environment you want to rotate keys for and you'll see a list of users
-and the age of their access keys.
+Log in to the [IAM Management Console] in the environment you want to rotate
+keys for and you'll see a list of users and the age of their access keys.
 
 ### 1. Find users access key
 
-Click on the user whose key you want to rotate. Click on `Security Credentials`. Make a note of the Access Key ID.
+- Click on the user whose key you want to rotate.
+- Click on the `Security Credentials` tab.
 
-### 2. Find where the key is used
+Users should always only have one access key. The only time a user should have
+two keys is when we are in the process of rotating the old one. For this reason,
+AWS allow users to have a maximum of two Access Keys at the same time.
 
-Enter [govuk-secrets/puppet](https://github.com/alphagov/govuk-secrets/tree/master/puppet) in your machine.
-Run one of the [Puppet AWS common actions](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws#common-actions) and search for the `Access Key ID` in the result.
+If this user has no keys, consider deleting it if it's not needed.
 
-> **Note**
-> The command you should run is related to the environment you need to target.
-> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.
+If this user has only one key, make a note of the `Access Key ID` and proceed to
+[find out if the key is used].
+
+If this user has two keys, you'll need to follow this guide twice. The first
+time, you'll need to consider one key as the old one to rotate and the other key
+as if it was a new one just created. The second time, you'll follow the guide as
+normal because the user will then have only one key. Now, choose which key is
+the one you are going to rotate and apply the following steps to that key.
```

Comment:
I think this is pretty much right, but could maybe be slightly clearer.

I'd perhaps stress that the first task is to remove one of the keys. If one isn't in use, then that's easy, as you just remove that key. If they're both in use, then you need to pick one, and then replace uses of it with the other key. Then, once it's no longer in use, you can remove it. I'm not sure that referring generally to other parts of this document will be helpful, as I think it's more likely to cause confusion.

Reply:
I agree, I'm not happy with this part myself and I re-wrote it several times in different ways. In the end, I decided to put down something anyway and see other's inputs.

I find it difficult to convey this message. I know what needs to be done but it isn't easy to explain, especially because to really understand this point, the reader should either be already familiar with the process of rotating keys or read this whole guide.

How about this? 👇

```
If this user has two keys, the first task is to remove one of them. If one isn't
in use, you can remove that key and proceed to rotate the other. If they're both
in use, you need to pick one and replace all uses of it with the other key.
Then, once it's no longer in use, you can remove it and proceed to rotate the
other.
```

Reply:
I think that's better.

I'm uncertain about the way this mentions rotating specific keys though: "proceed to rotate the other". I guess I know that I think this means that you should continue reading and following the documentation, as that's the subject of the document, but I think it could be read as "rotate" is something you need to do now, as part of this step.

But while I'd prefer not saying "rotate", and perhaps just say instead "proceed to the next step.", I think it's OK this way as well.